Cabal/doc/developing-packages.rst

```diff
@@ -1732,15 +1731,46 @@ system-dependent values for these fields.
           foo ^>= 1.2.3.4,
           bar ^>= 1
 
-    This allows to express the intent that this packages requires
-    versions of ``foo`` and ``bar`` which are semantically compatible
-    to ``foo-1.2.3.4`` and ``bar-1`` respectively. This subtle but important
-    difference in signaling allows tooling to treat *"hard"* ``<``-style
-    and *"weak"* ``^>=``-style upper bounds differently. For instance,
-    :option:`--allow-newer`'s ``^``-modifier allows to relax only *"weak"*
-    ``^>=``-style bounds while leaving ``<``-bounds unaffected.
+    This allows to assert the positive knowledge that this package is
+    *known* to be semantically compatible with the releases
+    ``foo-1.2.3.4`` and ``bar-1`` respectively. The information
+    encoded via such ``^>=``-assertions is used by the cabal solver to
+    infer version constraints describing semantically compatible
+    version ranges according to the PVP_ contract (see below).
+
+    Another way to say this is that ``foo < 1.3`` expresses *negative*
+    information, i.e. "``foo-1.3`` or ``foo-1.4.2`` will *not* be
+    compatible"; whereas ``foo ^>= 1.2.3.4`` asserts the *positive*
+    information that "``foo-1.2.3.4`` is *known* to be compatible" and (in
+    the absence of additional information) according to the PVP_
+    contract we can (positively) infer right away that all versions
+    satisfying ``foo >= 1.2.3.4 && < 1.3`` will be compatible as well.
 
-    Ignoring the signaling intent, the equivalences are
+    .. Note::
+
+       More generally, the PVP_ contract implies that we can safely
+       relax the lower bound to ``>= 1.2``, because if we know that
+       ``foo-1.2.3.4`` is semantically compatible, then so is
+       ``foo-1.2``. But we'd need to perform additional static
```

Comment:
that's not true, i think you meant to say `foo-1.2.3` here and below.

Reply:
I really intended to say `foo-1.2` here, as this Note is describing the property "if foo-1.2.3.4 compiles & is semantically correct/compatible; then *if* foo-1.2 compiles as well, then foo-1.2 is semantically correct/compatible as well". What you're referring to is the awkwardness, that `^>= 1.2.3.4` doesn't desugar to `>= 1.2.3 && < 1.3` (i.e. dropping the patch-level).